packages/yoroi-extension/app/containers/swap/categories/SwapPage.js

```diff
@@ -15,144 +13,124 @@ import { PublicDeriver } from '../../../api/ada/lib/storage/models/PublicDeriver
-      |},
-    |},
-  |} {
-    if (this.props.generated !== undefined) {
-      return this.props.generated;
-    }
-    if (this.props.stores == null || this.props.actions == null) {
-      throw new Error(`${nameof(SwapPage)} no way to generated props`);
-    }
-    const { stores, actions } = this.props;
-    const profileStore = stores.profile;
-    return Object.freeze({
-      stores: {
-        app: {
-          currentRoute: stores.app.currentRoute,
-        },
-        profile: {
-          setProfileLocaleRequest: {
-            isExecuting: profileStore.setProfileLocaleRequest.isExecuting,
-            error: profileStore.setProfileLocaleRequest.error,
-          },
-          LANGUAGE_OPTIONS: profileStore.LANGUAGE_OPTIONS,
-          currentLocale: profileStore.currentLocale,
-          currentTheme: profileStore.currentTheme,
-          UNIT_OF_ACCOUNT_OPTIONS: profileStore.UNIT_OF_ACCOUNT_OPTIONS,
-          unitOfAccount: profileStore.unitOfAccount,
-          setUnitOfAccountRequest: {
-            error: profileStore.setUnitOfAccountRequest.error,
-            isExecuting: profileStore.setUnitOfAccountRequest.isExecuting,
-          },
-        },
-        wallets: {
-          selected: stores.wallets.selected,
-          publicDerivers: stores.wallets.publicDerivers,
-          getLastSelectedWallet: stores.wallets.getLastSelectedWallet,
-        },
-        coinPriceStore: {
-          getCurrentPrice: stores.coinPriceStore.getCurrentPrice,
-          lastUpdateTimestamp: stores.coinPriceStore.lastUpdateTimestamp,
-          refreshCurrentUnit: {
-            isExecuting: stores.coinPriceStore.refreshCurrentUnit.isExecuting,
-          },
-        },
-      },
-      actions: {
-        wallets: {
-          setActiveWallet: { trigger: actions.wallets.setActiveWallet.trigger },
-        },
-        profile: {
-          updateLocale: { trigger: actions.profile.updateLocale.trigger },
-          updateTheme: { trigger: actions.profile.updateTheme.trigger },
-          updateUnitOfAccount: { trigger: actions.profile.updateUnitOfAccount.trigger },
-        },
-        router: {
-          goToRoute: { trigger: actions.router.goToRoute.trigger },
-        },
-      },
-    });
-  }
+      {/* Clear and switch */}
+      <Box display="flex" alignItems="center" justifyContent="space-between">
+        <Box sx={{ cursor: 'pointer' }}>
+          <SwitchIcon />
+        </Box>
+        <Box>
+          <Button variant="tertiary" color="primary">
+            Clear
+          </Button>
+        </Box>
+      </Box>
+
+      {/* To Field */}
+      <SwapInput label="Swap to" asset={{ amount: 0, ticker: '' }} />
+
+      {/* Price between assets */}
+      <Box mt="16px">
+        <PriceInput
+          assets={[
+            { ticker: 'TADA', amount: 20 },
+            { ticker: 'USDA', amount: 5 },
+          ]}
+          readonly={isMarketOrder}
+          label="Market price"
```

Comment:
I would suggest using this props structure instead. 

```suggestion
          baseAsset={{ ticker: 'TADA', amount: 20 }}
          quoteAsset={{ ticker: 'USDA', amount: 5 }}
          readonly={isMarketOrder}
          label="Market price"
```